fontc/src/main.rs

```diff
@@ -255,8 +246,63 @@ impl ChangeDetector {
     }
 }
 
+/// A unit of executable work plus the identifiers of work that it depends on
+struct Job {
+    // The actual task
+    work: Box<dyn Work + Send>,
+    // Things that must happen before us, e.g. our dependencies
+    happens_after: HashSet<WorkIdentifier>,
+}
+
+fn create_jobs(
```

Comment:
this could be some kind of constructor method on our imagined `JobSet` type? 🤔

Reply:
I left this as I like the model of compiler populates workload then runs it. If you have a chance take a look at where I ended and lmk what you think. I'm going to merge but I'm happy to make further changes based on additional feedback.